File: Math/1513. Number of Substrings With Only 1s.cpp
class Solution {
public:
    int numSub(string s) 
    {
        int mod = 1e9 + 7;
        long ans = 0, count = 0, n = s.size();
        for(int i = 0; i <= n; i++)
        {
            if(s[i] == '1') count += 1;
            else 
            {
                ans += (count * (count + 1) / 2) % mod;
                count = 0;
            }
        }
        return ans % mod;
    }
};

File: Math/1513. Number of Substrings With Only 1s.py
class Solution:
    def numSub(self, s: str) -> int:
        mod = int(1e9 + 7)

        s += '0'
        count, ans = 0, 0
        for ch in s:
            if ch == '1':
                count += 1
            else:
                ans += (count * (count + 1) / 2) % mod
                count = 0
        return int(ans % mod)
